code/datums/signals.dm

```diff
@@ -120,7 +120,9 @@
 	// all the objects that are receiving the signal get the signal this final time.
 	// AKA: No you can't cancel the signal reception of another object by doing an unregister in the same signal.
 	var/list/queued_calls = list()
-	for(var/datum/listening_datum as anything in target)
-		queued_calls[listening_datum] = listening_datum._signal_procs[src][sigtype]
-	for(var/datum/listening_datum as anything in queued_calls)
-		. |= call(listening_datum, queued_calls[listening_datum])(arglist(arguments))
+	// This should be faster than doing `var/datum/listening_datum as anything in target` as it does not implicitly copy the list
+	for(var/i in 1 to length(target))
+		var/datum/listening_datum = target[i]
+		queued_calls.Add(listening_datum, listening_datum._signal_procs[src][sigtype])
```

Comment:
you... might? get something by preallocating queued_calls's length, but that'd likely only really matter for large lists which are not a common case, would prob slow us down normally

Reply:
to clarify I mean in context of list expansion, but these lists are small so it doesn't matter/would maybe hurt